api/v1alpha1/external_secrets_config_types.go

```diff
@@ -201,3 +215,40 @@ type CertProvidersConfig struct {
 	// +kubebuilder:validation:Optional
 	CertManager *CertManagerConfig `json:"certManager,omitempty"`
 }
+
+// ComponentName represents the different external-secrets components that can have network policies applied.
+type ComponentName string
+
+const (
+	// CoreController represents the external-secrets component
+	CoreController ComponentName = "ExternalSecretsCoreController"
+
+	// BitwardenSDKServer represents the bitwarden-sdk-server component
+	BitwardenSDKServer ComponentName = "BitwardenSDKServer"
+)
+
+// NetworkPolicy represents a custom network policy configuration for operator-managed components.
+// It includes a name for identification and the network policy rules to be enforced.
+type NetworkPolicy struct {
+	// name is a unique identifier for this network policy configuration.
+	// This name will be used as part of the generated NetworkPolicy resource name.
+	// +kubebuilder:validation:Required
+	// +required
+	Name string `json:"name"`
+
+	// componentName specifies which external-secrets component this network policy applies to.
+	// +kubebuilder:validation:Enum:=ExternalSecretsCoreController;BitwardenSDKServer
+	// +kubebuilder:validation:Required
+	ComponentName ComponentName `json:"componentName"`
+
+	// egress is a list of egress rules to be applied to the selected pods. Outgoing traffic
+	// is allowed if there are no NetworkPolicies selecting the pod (and cluster policy
+	// otherwise allows the traffic), OR if the traffic matches at least one egress rule
+	// across all of the NetworkPolicy objects whose podSelector matches the pod. If
+	// this field is empty then this NetworkPolicy limits all outgoing traffic (and serves
+	// solely to ensure that the pods it selects are isolated by default).
+	// The operator will automatically handle ingress rules based on the current running ports.
+	// +optional
```

Comment:
Better to set this as required too, since just name and componentName doesn't server any porpose.

```suggestion
	// +kubebuilder:validation:Required
```